Improve the creature movepool page user experience

Menthx opened the pull request:
Thank you for your contribution to the **Pokémon Studio** repo.

Before submitting this PR into the develop branch, please make sure:

- [x] Your code builds clean without any errors or warnings
- [x] You are following the [Code guidelines](../CodeGuidelines.md)
- [x] You tested your code to make sure it does what it is supposed to do

## Description

The move pool page on a pokemon page has been review, to improve the user experience about it.


## Tests to perform

- [x] Latest moves added are listed in a separated list, at the bottom of the list
- [x] List accordions are replaced by an unique component with tabs for each list
- [x] The main component as a fixed height to make sure we do not need to scroll to see the bottom of it or access the buttons. This is the list that display a scroll bar.
- [x] The main component header should always remain visible and only the content should scroll (already the case)
- [x] For multiple move addition, a side editor panel helps the user to select multiple moves and bulk add them
- [x] When the user save its project, the component is updated and the list is refreshed: moves in the "latest moves" part are added to the main part
- [x] User are able to use keyboard navigation to easily use the component
- [x] (Optional) Add a highlight animation to focus the last added move in the list

#486

Menthx committed:
feature: Improve the creature movepool page user experience

Add new translations for "last added moves" and "add multiple moves" in English and French; update movepool components to support new functionality for adding moves and displaying recently added moves.

feature: Enhance movepool functionality with multiple move addition

Implement a new component for adding multiple moves to a Pokémon's movepool, including UI updates and integration with existing components. Add translations for new features across multiple languages, ensuring a consistent user experience.

See add button when no move is found

fix(move): multiple bug fixed

fix: changed pokemon was adding last added move, effect reworked

feat(movepoolLevelTable): add functionnality to movepool table by level

- Improve renderEditMove to be used multiple time (maybe can be more refactor)
- Add import, crud to level table

fix(movepoolleveltable): make a lot of fixe thought tables

fix: review some change and improve disparity about tabs change and creature change

fix(traduction): transform by tutoting by only tutoring

fix(style): adjust some alignement

fix(style): largeur composant fixed and tabs accessibility added

Improve save detection

Menthx commented:
Notes pour moi-meme, il ne reste plus que l'ajout multiple via l'éditor à faire !

Palbolsky commented on src/views/pages/database/Pokemon.Movepool.page.tsx:
```diff
@@ -53,13 +72,58 @@ export const PokemonMovepoolPage = () => {
+            {indexTabs === 0 && (
+              <MovepoolLevelLearnableTable importation={{ label: t('movepool_import'), onClick: () => setCurrentEditor(movepoolType) }} />
+            )}
+            {indexTabs === 1 && (
```
A voir s'il est possible d'utiliser un petit objet / record pour mapper entre l'index et le type de movepool pour éviter la répétition de code.

Menthx replied:
Yes, j'ai commencé un regarder un truc simple, et je dois retravailller tout les tableau xqui utilise le même composant car il y a un soucis d'actualisation.
Normal, avec tous ces if qui sautent d'avoir d'autres soucis x)

Palbolsky commented on src/views/components/database/pokemon/movepool/MovepoolTable.tsx:
```diff
@@ -232,37 +275,193 @@ export const MovepoolTable = ({ movepoolType }: MovepoolTableProps) => {
+          </DataMoveGrid>
+          <ScrollableContent>
+            {movepoolData.map((learnableMove, index) => (
+              <RenderEditMove
```
Il manque une propriété key ici.

<img width="1054" height="227" alt="image" src="https://github.com/user-attachments/assets/d1c60ef9-2ac3-416b-baa1-cf8b57d4c82c" />

AerunDev reviewed (changes requested):
Salut @AntoinePoree ! 😄
Encore une fois, gros taff sur cette PR, bien joué à toi, c'est top de voir vos travaux de conception et de design mis en mouvement comme ça sur la PR, les utilisateurs vont vraiment apprécier.

J'ai noté quelques remarques, que je classe par domaine.
Désolé pour le nombre, mais pour la plupart c'est du design que je pense rapide à corriger. Je préfère le faire remonter maintenant et que le travail soit impeccable à la sortie et que ça sublime ce que vous avez fait ! 🎉 

## Design
- [x] La largeur du composant global doit être la même que celle du bouton segmenté "Pokémon | Movepool | Ressources". Comme ça, quand on passe d'un onglet à l'autre on a un design cohérent.
- [x] On peut peut-être gagner encore de la place pour avoir 32px seulement en dessous du composant.
- [x] Quand on survole une ligne de la liste, ou par exemple quand une ligne est en surbrillance après ajout, on voit que la partie de gauche du background est rognée au lieu de prendre encore 4px qu'elle devrait prendre en dépassant (cf. Figma).
- [x] Dans l'éditeur d'ajout multiple, le bouton de suppression d'une ligne a un comportement étrange au survol de la souris, comme un cercle foncé apparaît au centre.
- [x] J'ai souvenir qu'on en ait parlé, mais il y a un problème avec l'espacement entre les éléments de la liste des derniers ajouts : l'espacement est plus grand de deux fois celui de l'espacement entre les éléments de la liste déjà existante au dessus. Il faudrait qu'il soit le même.

## Accessibilité
- [x] La navigation au clavier n'est pas possible dans le menu de navigation entre les différentes listes.
- [x] On a un seul mot sur toutes les tabs sauf "Par maître des capacités". Ne peut-on pas mettre "Maître des capacités" si ça passe et que le texte n'est pas utilisé ailleurs ?

## Régression
- [x] À chaque fois que j'ajoute une capacité à un Fakemon, et ce même s'il a peu, ça affiche le loader à la sauvegarde. Ça me parait très lourd. S'agit-il d'un problème de scope de la sauvegarde ou d'un problème d'optimisation ?

Menthx commented:
Top, merci pour ces retours.
En ce qui concerne **l'accessibilité**, pour la navigation entre les onglets, je ne peux pas faire un shortcut global vu qu'il y a en a deux présent sur cette même page. Donc j'ai passé les tabs en accessibilité pour qu'il soit sélectionnable via shift et les flèches une fois dedans.

En ce qui concerne la **sauvegarde**, une utilisation innaproprié de "isDataToSave" provoquait le problème, on passe désormais par un event donc plus de loader de sauvegarde, sauf sur les changements conséquant !

Donc tu vas pouvoir regarder tout ça 💪

Menthx committed:
feat(movepool): add possibility to erase input level to increase UX

Palbolsky committed:
Fix style after Ralendel test

AerunDev reviewed (changes requested):
Yop ! Malheureusement les tests ont permis de trouver un bug qui peut être assez embêtant. J'en profite pour rappeler le deuxième qui était remonté, et qui sera traité plus tard si vous voulez.

## Capacité ajoutée qui n'apparaît pas

La première capacité apprise par montée de niveau n'est parfois pas ajoutée à la liste des ajouts récents, et ne déclenche pas la sauvegarde.

### Cas de reproduction

1. Ouvrir Pokémon Studio et charger un projet
2. Se rendre sur l'onglet "Movepool" de Bulbizarre
3. Sélectionner successivement "Maître des capacités, Technique, Reproduction, Évolution, puis revenir sur Reproduction, Technique, Maître des capacités et Niveau"
4. Aller sur la page de Chenipan pour afficher l'onglet "Movepool"
5. Sélectionner l'onglet "Technique" puis revenir sur "Niveau"
6. Dans la liste d'ajout d'une capacité en bas à gauche, choisir "Cru Ailes"
7. Ajouter la capacité via le bouton "Ajouter"
8. Voir la capacité apparaître une demi seconde avant de disparaitre sans activer la sauvegarde
9. Ajouter la capacité "Cru Ailes" à nouveau pour la voir apparaître

## La liste d'ajout rapide n'est pas mise à jour pour chaque onglet

Lorsqu'on passe des onglets Maître des capacités à Technique ou de Technique à Reproduction, la liste d'ajout rapide n'est pas mise à jour. Ce qui a pour résultat d'afficher l'entête "CS01 -" pour Coupe alors qu'on est en train d'éditer une liste qui n'est pas celle des Techniques, et inversement.

Menthx committed:
fix(movepool-table): some bugs now dissapears

- move who didnt appear
- movepool from render edit mvoe who didnt updaqte on tab change
